Modules/_abc.c

```diff
@@ -624,8 +624,7 @@ _abc__abc_instancecheck_impl(PyObject *module, PyObject *self,
 
     switch (PyObject_IsTrue(result)) {
     case -1:
-        Py_DECREF(result);
-        result = NULL;
+        Py_CLEAR(result);
```

Comment:
`Py_DECREF(var); var=NULL` only has a race condition if `var` is accessible from outside the function, right? I don't think there is any race condition if `var` is just a local variable.

Reply:
Yeah, for those cases you "only" get increased readability and fewer lines of code.

Reply:
You're right, the code is only unsafe if the variable is shared between two function calls. It can be a static variable inside the function, a global variable, an object attribute, etc. Here result is a local variable, so the code is safe. Do you think that the commit message should be enhanced?

@erlend-aasland is right, it's also about about increasing readability.